### ISSUE TITLE
Bug: Backdrop scroll behavior.

### ISSUE THREAD

**Issue by Domvel** [isaacplmann/ngx-tour#171]
If backdrop is enabled. (highlight element) It does not behave correctly on scroll. It does not move with the focused element.

Solution: 
- Disable scrolling while tour is enabled.
- Or place the backdrop (highlight) correctly. Move with the element like in angular-ui-tour.

## ngx-tour (bug)
![export-ngx-tour](https://user-images.githubusercontent.com/35452565/53960780-6ce3c400-40e7-11e9-9fa6-1a2df376a92d.gif)

## angular-ui-tour (expected)
![export-ui-tour](https://user-images.githubusercontent.com/35452565/53960781-6f461e00-40e7-11e9-8d15-ed53a9c5e50d.gif)

Bug discovered in [current demo](https://isaacplmann.github.io/ngx-tour) from 2019-03-07.
Compare with [angular-ui-tour](http://benmarch.github.io/angular-ui-tour/#/docs).

**Comment by hakimio** [comment]
@Domvel
- If you use "mat-sidenav-container", scrolling will be disabled by default when ui-tour is active
- Feel free to create a PR addressing the issue in other cases

**Comment by amitagrawal11** [comment]
@hakimio Please let us know which scrolling do we need to disable, Is it Page scrolling, Container Scrolling or tour-step-template scrolling ?

**Comment by amitagrawal11** [comment]
@hakimio Any updates on this one please ?

**Comment by hakimio** [comment]
@amitagrawal11 you should be able to see yourself which one of your divs/containers is scrolling when the tour is active by just inspecting the DOM. Solution is quite simple:
1) Before starting the tour add "no-scroll" class to that container
```css
.no-scroll {
    overflow: hidden;
}
```
```js
const body = document.body,
    mainSection = document.querySelector('.main-content');

mainSection.classList.add('no-scroll');
```
2) Add tour end event listener where you remove the class.
```js
this.tourService.end$.subscribe(() => {
   this.enableScroll();
});
```

It's a bit tricky to come-up with general solution for everyone since scrolling is implemented differently from one app to the other and some have multiple scrollable containers.

**Comment by amitagrawal11** [comment]
@hakimio thanks for your response, It worked for me now. 

But I can see another bug by disabling scrollbar onto container div i.e. suppose we have two tooltips on one page and first one is placed on the top of the page and second one is placed on bottom of the page. So problem is if I click on backtip/prevtip (while navigating from second tip to first tip) then page does not scroll and tip comes inconsistently.

**Comment by hakimio** [comment]
Scrolling the page programmatically still works fine when overflow is hidden. That's not an issue.

**Comment by JayEnaR** [comment]
Has this been resolved? I'm facing the same issue using `mat-drawer` using the latest version in Angular 17.

**Comment by hakimio** [comment]
If you can reproduce the issue with the latest version of [ngx-ui-tour](https://github.com/hakimio/ngx-ui-tour) library, open a new issue with reproduction in `ngx-ui-tour` repository.

**Comment by hakimio** [comment]
You can also just use `disablePageScrolling` config in the new library.
